src/components/shared/ManagedTree.js

```diff
@@ -121,15 +124,24 @@ class ManagedTree extends Component<Props, State> {
     const { expanded } = this.state;
     // This file is visible, so we highlight it.
     if (expanded.has(this.props.getPath(highlightItems[0]))) {
-      this.focusItem(highlightItems[0]);
-    } else {
-      // Look at folders starting from the top-level until finds a
-      // closed folder and highlights this folder
-      const index = highlightItems
-        .reverse()
-        .findIndex(item => !expanded.has(this.props.getPath(item)));
-      this.focusItem(highlightItems[index]);
+      return this.focusItem(highlightItems[0]);
     }
+
+    // Look at folders starting from the top-level until finds a
+    // closed folder and highlights this folder
+    const reversedItems = highlightItems.reverse();
+    let index = reversedItems.findIndex(
+      item => !expanded.has(this.props.getPath(item))
+    );
+
+    if (this.props.autoExpandOnHighlight) {
+      while (index <= reversedItems.length - 2) {
+        this.setExpanded(reversedItems[index], true, false);
+        index++;
+      }
```

Comment:
hmm, can you discuss what this does?

Reply:
I have added a comment to make it more clear